components/config_plugins/lib.rs

```diff
@@ -199,23 +196,32 @@ impl Field {
 }
 
 fn attr_to_pref_name(attr: &Attribute) -> Option<LitStr> {
-    attr.parse_meta().ok().and_then(|meta| {
-        if let Meta::List(MetaList { path, nested, .. }) = meta {
-            if path.is_ident("serde") {
-                if let Some(NestedMeta::Meta(Meta::NameValue(MetaNameValue {
-                    ref path,
-                    lit: Lit::Str(val),
-                    ..
-                }))) = nested.iter().next()
-                {
-                    if path.is_ident("rename") {
-                        return Some(val.clone());
-                    }
-                }
+    if attr.path().is_ident("serde") {
+        let mut res = None;
+        // if this fails result is still None
+        let _ = attr.parse_nested_meta(|meta| {
+            if meta.path.is_ident("rename") {
+                res = Some(meta.value()?.parse()?);
             }
-        }
+            Ok(())
+        });
+        res
+    } else {
         None
-    })
+    }
+    /*let mut res = None;
+    attr.parse_nested_meta(|meta| {
+        if meta.path.is_ident("serde") {
+            meta.parse_nested_meta(|meta| {
+                if meta.path.is_ident("rename") {
+                    res = Some(meta.value()?.parse()?);
+                }
+                Ok(())
+            })?;
+        }
+        Ok(())
+    }).unwrap();
+    return res;*/
```

Comment:
Is there a reason this code is commented out but still here?

Reply:
That should not be here.